### PR TITLE
remove unnecessary zlib submodule

### PR DESCRIPTION
**Pull request by Green-Sky**
Pretty self explanatory.
DISCLAIMER: I did not test this extensively, but as far as my google-foo goes, `Z_DEFAULT_COMPRESSION` and `MZ_COMPRESS_LEVEL_DEFAULT` are both defined as `-1`.

### CODE REVIEW

**Comment by Sygmei** [comment]
This won't work, without zlib compressing isn't possible
It worked in your case because minizip automatically fetches zlib (using `MZ_FETCH_LIBS` flag) which is a less desirable behaviour compared to cloning with submodules from the beginning imho.
UnitTests breaks when zlib is not available : 
```
[  FAILED  ] UnzipFileTestFixture.files_are_actualy_created_while_unzipping (77 ms)
[ RUN      ] UnzipFileTestFixture.unziped_files_are_equivalent_to_original
unknown file: error: C++ exception with description "exception occurred when extracting file 'UnitTests.exe' : dump error" thrown in the test body.
```

**Comment by Sygmei** [comment]
After a few tests, it seems that minizip doesn't care about the included version of zlib anyway.
Also found that minizip automatically pulls quite a lot of unwanted libraries, I pushed the changes from your PR to master.
@Assertores do you have an idea on how to make minizip use the bundled zlib ?

**Comment by Assertores** [comment]
if i understand the cmake of minizip correctly, it will check for a zlib version as long as `MZ_ZLIB` is on (which it is by default). and as long as `MZ_FETCH_LIBS` is active (which it is for windows by default) it will automaticly fetch it's on zlib version.

TLDR: minizip should use it's bundled zlib version by default.

(i hope i understood your question correctly)

**Comment by Sygmei** [comment]
@Assertores thanks for taking a look at it :)

I will keep the behaviour as-is for now, zlib will be downloaded by minizip's CMakeLists.txt.

I can close this PR as the changes have been manually merged to master, thanks @Green-Sky for finding out this weird behaviour !
